bazaar/src/lib.rs

```diff
@@ -185,3 +185,14 @@ decl_module! {
         }
     }
 }
+
+impl<T: Config> Module<T>
+{
+    pub fn get_businesses(cid: &CommunityIdentifier) -> Vec<(T::AccountId, BusinessData)> {
```

Comment:
Hmm, I haven't thought about this before: We iterate through the whole storage here. I think we should either mark the rpc as unsafe, i.e., only expose the rpc if the node is started with `--unsafe-rpc-external` (exact flag name might be different), or we cache the Vec offchain, similar to how we do it in the communities api. Such that subsequent calls will be less expensive.

I don't know about the latter though because caching is only useful, if we don't mark it dirty too often...

Maybe use `DenyUnsafe`?`

Reply:
I am unsure about the effects of such unsafe markings. Caching businesses might be discussable, doing it for offerings might be too optimistic.

Reply:
The effect is minimal, it does nothing besides not accepting the rpc, when the unsafe flag isn't set.